docs/appendix/upgrade-notes/3.x-to-4.0.rst

```diff
@@ -23,6 +23,12 @@ Because of the `deprecation`_ of the ``Zend\Mail`` library and it's replacement
 
 .. _deprecation: https://www.zend.com/blog/evolution-zend-framework-laminas-project
 
+Doctrine DBAL
+-------------
+
+Elgg replaced v2 with v3 of the ``doctrine/dbal`` dependency. The most notable change is that if you work with QueryBuilders and use the ``$qb->fetch()``
+function you will no longer get an object, but an array. If you want your rows to be useable as classes, you can use ``elgg()->db->getData($qb)``.
+
```

Comment:
Should we mention the change in key format? Even if it’s API we don’t encourage people to use, it could break some plugins.

Reply:
which change are you referring to?

Reply:
oh you mean the key change for the params... yep... can add some hint for that

Reply:
it has been mentioned